### PR TITLE
Fix Run info badge for running runs

### LINKED ISSUE #655: Continuing with Experiment Run fails to show "active" on the run page.
This is a run that has been restarted with `with mlflow.start_run(run_id=run_id) as run:`

https://photos.app.goo.gl/9K8Riq92T9W4fnfAA it shows active on the main page

But not on the runs page: https://photos.app.goo.gl/HVVgYrTk3H69WGoN6

Also:
- The time active for the run does not start increasing
- The activity log for the experiment does not show any events

The important stuff however, like the metrics, still work so this is not too big a problem. Thank you

### PR DESCRIPTION
**Pull request by fabiovincenzi**
Fixes #655 
This PR addresses the issue related to the display of the active state of runs after a restart.
AIM UI shows the status badge based on the Endtime of a run so we need to set Endtime to null when restarted.

### CODE REVIEW

File: pkg/api/mlflow/dao/convertors/run.go [hunk 1]
```diff
@@ -86,9 +86,13 @@ func ConvertCreateRunRequestToDBModel(
...
-	run.EndTime = sql.NullInt64{
-		Int64: req.EndTime,
-		Valid: true,
+	if req.Status == string(models.StatusRunning) {
+		run.EndTime = sql.NullInt64{}
+	} else {
+		run.EndTime = sql.NullInt64{
+			Int64: req.EndTime,
+			Valid: true,
+		}
```
Comment (dsuhinin) [inline pkg/api/mlflow/dao/convertors/run.go:95]: can you please add a comment here so we can always know why we need this `if` statement.

File: pkg/api/mlflow/dao/repositories/run.go [hunk 2]
```diff
@@ -292,6 +292,12 @@ func (r RunRepository) UpdateWithTransaction(ctx context.Context, tx *gorm.DB, r
 	if err := tx.WithContext(ctx).Model(&run).Updates(run).Error; err != nil {
 		return eris.Wrapf(err, "error updating existing run with id: %s", run.ID)
 	}
+	// Use UpdateColumn so we can reset EndTime to null
+	if run.EndTime.Int64 == 0 {
```
Comment (dsuhinin) [inline pkg/api/mlflow/dao/repositories/run.go:296]: EndTime has `IsValid` so in most cases it is much understandable to use it.

File: pkg/api/mlflow/dao/repositories/run.go [hunk 3]
```diff
@@ -292,6 +292,12 @@ func (r RunRepository) UpdateWithTransaction(ctx context.Context, tx *gorm.DB, r
...
+		if err := tx.Model(&run).UpdateColumn("EndTime", sql.NullInt64{}).Error; err != nil {
```
Comment (dsuhinin) [inline pkg/api/mlflow/dao/repositories/run.go:297]: we specially use `UpdateColumn` to just make field nullable, right?

**Review by dsuhinin — CHANGES_REQUESTED** [review]
I can't see a test for this special case. If it possible it would be really nice to have it.

File: pkg/api/mlflow/dao/repositories/run.go (hunk 3 above)
Comment (fabiovincenzi) [reply pkg/api/mlflow/dao/repositories/run.go:297]: Exactly, regular `Updates` function skip nil values

Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/run.go:297]: even if you pass like `sql.NullInt64{IsValid: true}`?

Comment (fabiovincenzi) [reply pkg/api/mlflow/dao/repositories/run.go:297]: It actually works! Thanks!

Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/run.go:297]: ohh, that's nice! thank you!

File: tests/integration/golang/mlflow/run/update_test.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ func TestUpdateRunTestSuite(t *testing.T) {
 }
 
 func (s *UpdateRunTestSuite) Test_Ok() {
-	// create test run.
+	// create test runs.
```
Comment (dsuhinin) [inline tests/integration/golang/mlflow/run/update_test.go:31]: but you create one run?

**Review by dsuhinin — APPROVED** [review]
Looks good!

File: tests/integration/golang/mlflow/run/update_test.go (hunk 4 above)
Comment (fabiovincenzi) [reply tests/integration/golang/mlflow/run/update_test.go:31]: they are two

Comment (dsuhinin) [reply tests/integration/golang/mlflow/run/update_test.go:31]: aaa, understood.

**Pull request merged**